<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>790</width>
    <height>619</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>splot</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QFrame" name="verticalFrame_2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QFrame" name="sourceFrame_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTipDuration">
          <number>-1</number>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Plain</enum>
         </property>
         <layout class="QFormLayout" name="sourceFrame">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="formAlignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="sourceLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Source:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="serialPortComboBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select a serial port attached to the system.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;You can also enter an IP address and port, e.g. &amp;quot;127.0.0.1:19021&amp;quot; to connect to port 19021 on the local machine. By default, splot assumes you want a tcp connection, but you can receive udp data as well by entering &amp;quot;udp://&amp;lt;IP&amp;gt;:port&amp;quot;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="toolTipDuration">
             <number>-1</number>
            </property>
            <property name="editable">
             <bool>true</bool>
            </property>
            <property name="sizeAdjustPolicy">
             <enum>QComboBox::AdjustToContents</enum>
            </property>
            <property name="frame">
             <bool>true</bool>
            </property>
            <item>
             <property name="text">
              <string>(not connected)</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="dataRateBpsLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Data rate:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="dataRateBpsValueLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>---</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QFrame" name="leftFrame">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>250</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>250</width>
             <height>16777215</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QFrame" name="serialParametersFrame">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QFormLayout" name="serialParametersFormLayout">
               <property name="fieldGrowthPolicy">
                <enum>QFormLayout::ExpandingFieldsGrow</enum>
               </property>
               <property name="verticalSpacing">
                <number>5</number>
               </property>
               <property name="leftMargin">
                <number>12</number>
               </property>
               <property name="topMargin">
                <number>10</number>
               </property>
               <property name="rightMargin">
                <number>12</number>
               </property>
               <property name="bottomMargin">
                <number>10</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="serialBaudRateLabel">
                 <property name="text">
                  <string>Baud rate</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="serialBaudRateComboBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="editable">
                  <bool>true</bool>
                 </property>
                 <property name="insertPolicy">
                  <enum>QComboBox::NoInsert</enum>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="serialParityLabel">
                 <property name="text">
                  <string>Parity</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QComboBox" name="serialParityComboBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="serialStopBitsLabel">
                 <property name="text">
                  <string>Stop bits</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QComboBox" name="serialStopBitsComboBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="serialReadChunkSizeLabel">
                 <property name="text">
                  <string>Read chunk size</string>
                 </property>
                </widget>
               </item>
               <item row="5" column="0">
                <widget class="QLabel" name="serialBufferSizeLabel">
                 <property name="text">
                  <string>Serial buffer size</string>
                 </property>
                </widget>
               </item>
               <item row="5" column="1">
                <widget class="QSpinBox" name="serialBufferSizeSpinBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="keyboardTracking">
                  <bool>false</bool>
                 </property>
                 <property name="minimum">
                  <number>10</number>
                 </property>
                 <property name="maximum">
                  <number>99999999</number>
                 </property>
                 <property name="value">
                  <number>100000</number>
                 </property>
                </widget>
               </item>
               <item row="4" column="1">
                <widget class="QSpinBox" name="serialReadChunkSizeSpinBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="keyboardTracking">
                  <bool>false</bool>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>99999999</number>
                 </property>
                 <property name="value">
                  <number>10000</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="frame">
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QFormLayout" name="formLayout">
               <property name="fieldGrowthPolicy">
                <enum>QFormLayout::ExpandingFieldsGrow</enum>
               </property>
               <property name="formAlignment">
                <set>Qt::AlignRight|Qt::AlignTop|Qt::AlignTrailing</set>
               </property>
               <property name="verticalSpacing">
                <number>5</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="label_13">
                 <property name="text">
                  <string>Data format</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="dataFormatComboBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <item>
                  <property name="text">
                   <string>binary</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>ascii</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="messageDelimiterLabel">
                 <property name="text">
                  <string>Separator</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="messageDelimiterLineEdit">
                 <property name="text">
                  <string>0</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="binaryDtypeStringLabel">
                 <property name="text">
                  <string>Binary format</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLineEdit" name="binaryDtypeStringLineEdit">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Binary messages are parsed by numpy as structured datatypes (&lt;a href=&quot;https://numpy.org/doc/stable/user/basics.rec.html#structured-datatypes&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;https://numpy.org/doc/stable/user/basics.rec.html#structured-datatypes&lt;/span&gt;&lt;/a&gt;), which can be specified as comma-separated strings (see also &lt;a href=&quot;https://numpy.org/doc/stable/reference/arrays.dtypes.html#arrays-dtypes-constructing&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;https://numpy.org/doc/stable/reference/arrays.dtypes.html#arrays-dtypes-constructing&lt;/span&gt;&lt;/a&gt;). For now, it is assumed that each element of the datatype is a scalar, so nested/hierarchical dtypes may produce funky results and errors. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="toolTipDuration">
                  <number>-1</number>
                 </property>
                 <property name="text">
                  <string>u1,u1,u2,u2,u2</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="numberOfStreamsLabel">
                 <property name="text">
                  <string>Number of streams</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QSpinBox" name="numberOfStreamsSpinBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="keyboardTracking">
                  <bool>false</bool>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QFrame" name="frame">
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QFormLayout" name="formLayout_4">
               <property name="fieldGrowthPolicy">
                <enum>QFormLayout::ExpandingFieldsGrow</enum>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="plotLengthLabel">
                 <property name="text">
                  <string>Plot length</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="plotLengthSpinBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="keyboardTracking">
                  <bool>false</bool>
                 </property>
                 <property name="minimum">
                  <number>10</number>
                 </property>
                 <property name="maximum">
                  <number>100000</number>
                 </property>
                 <property name="value">
                  <number>1000</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="0" colspan="2">
                <widget class="QPushButton" name="pausePushButton">
                 <property name="text">
                  <string>Pause</string>
                 </property>
                 <property name="checkable">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="verticalFrame">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <layout class="QVBoxLayout" name="plotVBoxLayout"/>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>790</width>
     <height>37</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
